File: MASICPeakFinder/clsBaselineNoiseStatsSegment.vb
Public Class clsBaselineNoiseStatsSegment

    Public Property BaselineNoiseStats As clsBaselineNoiseStats

    Public Property SegmentIndexStart As Integer

    Public Property SegmentIndexEnd As Integer

    <Obsolete("Use the constructor that takes an instance of clsBaselineNoiseStats")>
    Public Sub New()
        BaselineNoiseStats = New clsBaselineNoiseStats()
    End Sub

    Public Sub New(noiseStats As clsBaselineNoiseStats)
        BaselineNoiseStats = noiseStats
    End Sub
End Class

File: MASICPeakFinder/clsBaselineNoiseStatsSegment.cs
using System;

namespace MASICPeakFinder
{
    /// <summary>
    /// Baseline noise stats segment
    /// </summary>
    public class clsBaselineNoiseStatsSegment
    {
        /// <summary>
        /// Baseline noise stats
        /// </summary>
        public clsBaselineNoiseStats BaselineNoiseStats { get; set; }

        /// <summary>
        /// Segment start index
        /// </summary>
        public int SegmentIndexStart { get; set; }

        /// <summary>
        /// Segment end index
        /// </summary>
        public int SegmentIndexEnd { get; set; }

        /// <summary>
        /// Obsolete constructor
        /// </summary>
        [Obsolete("Use the constructor that takes an instance of clsBaselineNoiseStats")]
        public clsBaselineNoiseStatsSegment()
        {
            BaselineNoiseStats = new clsBaselineNoiseStats();
        }

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="noiseStats"></param>
        public clsBaselineNoiseStatsSegment(clsBaselineNoiseStats noiseStats)
        {
            BaselineNoiseStats = noiseStats;
        }
    }
}